zion/src/main/scala/edu/uci/ics/cloudberry/zion/actor/DataStoreManager.scala

```diff
@@ -161,12 +161,33 @@ class DataStoreManager(metaDataset: String,
     val dropTableName = dropTable.dataset
 
     if(metaData.contains(dropTableName)){
-      metaData.remove(dropTableName)
+      context.child("data-" + dropTableName) match {
+        case Some(datasetActor) => datasetActor ! PoisonPill
+        case None =>
+      }
 
-      //TODO send query to DELETE dataset in database
+      val metaRecordFilter = FilterStatement(DataSetInfo.MetaSchema.fieldMap("name"), None, Relation.matches, Seq(dropTableName))
+      metaActor ! DeleteRecord(metaDataset, Seq(metaRecordFilter))
```

Comment:
to make logic modularized, after 170, we should remove `dropTableName` from `metaData` instead till 184.